src/backend/processor/diagnostics.ts

```diff
@@ -0,0 +1,196 @@
+
+            if (diagnostic) {
+                return {
+                    position: this._selectedEntry!,
+                    diagnostic
+                };
+            }
+        }
+
+        return undefined;
+    }
+
+    public setActiveReport(position: {file: string, idx: number} | undefined) {
+        if (position) {
+            const activeFile = this._diagnosticEntries.get(position.file);
+            const diagnostic = activeFile?.diagnostics[position.idx];
+
+            if (diagnostic) {
+                this._selectedEntry = position;
+                return;
+            }
+        }
+
+        this._selectedEntry = undefined;
+    }
+
+    constructor(ctx: ExtensionContext) {
+        ctx.subscriptions.push(this._diagnosticsUpdated = new EventEmitter());
+        window.onDidChangeVisibleTextEditors(this.onDocumentsChanged, this, ctx.subscriptions);
+        ExtensionApi.metadata.metadataUpdated(this.onMetadataUpdated, this, ctx.subscriptions);
+
+        this.init();
+    }
+
+    private init() {
+        // Initialize the currently opened windows
+        this.onDocumentsChanged(window.visibleTextEditors);
+    }
+
+    /** Prefer calling ``reloadDiagnosticsAsync`` in an async context */
+    reloadDiagnostics(forceReload?: boolean) {
+        this.reloadDiagnosticsAsync(forceReload)
+            .catch((err) => {
+                console.error(err);
+                window.showErrorMessage('Unexpected error when reloading diagnostics \nCheck console for more details');
+            });
+    }
+
+    // TODO: Add support for cancellation tokens
+    async reloadDiagnosticsAsync(forceReload?: boolean): Promise<void> {
+        // TODO: Allow loading all diagnostics at once
+        const plistFilesToLoad = this._openedFiles.flatMap(file => ExtensionApi.metadata.sourceFiles.get(file) || []);
+
+        if (this._selectedEntry) {
+            plistFilesToLoad.push(this._selectedEntry.file);
+        }
+
+        const loadedPlistFiles = new Set<string>();
+        const newEntries = forceReload
+            ? new Map()
+            : new Map(this._diagnosticEntries.entries());
+        const newSourceFiles = new Map();
+
+        let plistErrors = false;
+
+        // Load new .plist files
+        for (const plistFile of plistFilesToLoad) {
+            if (newEntries.has(plistFile)) {
+                loadedPlistFiles.add(plistFile);
+                continue;
+            }
+
+            try {
+                const diagnosticEntry = await parseDiagnostics(plistFile);
+                newEntries.set(plistFile, diagnosticEntry);
+                loadedPlistFiles.add(plistFile);
+            } catch (err) {
+                switch (err.code) {
+                // Silently ignore file-related errors
+                case 'FileNotFound':
+                case 'Unavailable':
+                    break;
+
+                default:
+                    console.error(err);
+                    plistErrors = true;
```

Comment:
I think we can remove the `plistErrors` variable because we set it to true only here and later we print an error message based on this variable:
```suggestion
                    window.showErrorMessage('Failed to read some CodeChecker diagnostic data\nCheck console for more details');
```

Reply:
The plistErrors flag is used to prevent showing multiple motifications of the same type, for example when 2 diagnostic files are corrupted.
In that case, both errors' stack traces are logged to the console, but only one notification is shown to the user inside Code.